### DISCUSSION TITLE
open config using $EDITOR

### DISCUSSION THREAD

**Discussion by AnirudhG07** [ghostty-org/ghostty#2961] [Feature Requests, Ideas]
I am very excited to use Ghostty, I have VERY recently got it. One thing that I am fond of using in my MacOS is using `command + comma` to open settings or config files. In iTERM, they had a gui, for Kitty they used vim(which I changed to nvim). But Ghostty uses TextEdit. 
I discussed this on discord and people said it is not possible yet. Can it be possible in future so that the config file can be open with $EDITOR, or similar. Maybe in the default configs, you can add an option of opening the config file in EDITOR. 

Thanks for Ghostty! Loving it!
(Sorry if this is repeated, although I coudnt find a discussion for it.)

**Comment by ralgozino** [comment]
IMHO <kbd>Command</kbd>+<kbd>,</kbd> should open the (non existing yet) settings GUI.

May be a different keybinding could be use to launch $EDITOR instead, but I think that having a flag like `+show-config --path` could be enough, so you can do `vim $(ghostty +show-config --path)`.

**Reply by AnirudhG07** [reply]
Yes true, it is currently opening the config, which is in a way settings only.
Thanks for the cmd

**Reply by mitchellh** [reply]
> IMHO Command+, should open the (non existing yet) settings GUI.

That's the plan, there is another discussion tracking that.

**Comment by telemachus** [comment]
> IMHO <kbd>Command</kbd>+<kbd>,</kbd> should open the (non existing yet) settings GUI.

Agreed: this is standard on macOS apps.

**Comment by noahfraiture** [comment]
Hey, also command like write_screen_file:open open in default macos file editor instead of $EDITOR. Anyway to change that ? This is not a standard in macos like command + ,

**Reply by 00-kat** [reply]
It's not yet implemented; that's the point of this feature request's existence.

**Comment by rhodes-b** [comment]
ghostty supports this now `ghostty +edit-config`

**Reply by SethMilliken** [reply]
How can `ghostty +edit-config` be used as the rhs for a `keybind`?

**Reply by 00-kat** [reply]
I don't think it can be.

**Reply by SethMilliken** [reply]
Sounds like this isn’t actually addressed yet, then, where _this_ is: “Use `Cmd+,` to open config in $EDITOR instead of TextEdit”.

**Reply by 00-kat** [reply]
Fair enough; reopened.

**Reply by andelink** [reply]
> How can `ghostty +edit-config` be used as the rhs for a `keybind`?

@SethMilliken see my comment here: https://github.com/ghostty-org/ghostty/discussions/2961#discussioncomment-16304191

**Reply by SethMilliken** [reply]
This technique makes a couple of assumptions that are rarely true for me. The frontmost Ghostty window must be `localhost`, and there must be an otherwise unoccupied prompt waiting (i.e. not in vim, less, mutt, top, etc.).

The point here is to have a keybinding that results in consistent behavior regardless of the characteristics of the frontmost window.

**Comment by 00-kat** [comment]
xref #2504.

**Comment by andelink** [comment]
I enable this with the following keybind in my ghostty config:

```ini
# Override the native 'edit config' shortcut with the CLI action, respecting ${VISUAL:-${EDITOR}}
#   1. `\x1bq` runs the zsh `push-line-or-edit` widget (bound to alt-q)
#   2. `ghostty +edit-config` is sent to the shell prompt
#   3. `\x0d` enters/executes the command currently on the prompt
keybind = unconsumed:super+,=text:\x1bqghostty +edit-config\x0d
```

You can use `bindkey` to view/modify your zsh key bindings as needed:
```zsh
❯ bindkey -L | grep push-line-or-edit
bindkey "^[q" push-line-or-edit  # command to bind it to alt-q
```

**Reply by radoslaw-panuszewski** [reply]
It does the trick, thank you! 🔥
